cegs_portal/search/templates/search/v1/experiment.html

```diff
@@ -348,10 +352,348 @@
+
+    // now you can use the API!
+    fetch("http://127.0.0.1:8000/static/search/experiments/{{ experiment.accession_id }}/g_per_c.bin")
+        .then((response) => response.arrayBuffer())
+        .then((buffer) => {
+            const start = performance.now();
+            let data = new DataView(buffer);
+            let offset = 0;
+            let values = [];
+            let binSize = data.getUint32(offset);
+            let mean = data.getFloat32(offset + 4);
+            let median = data.getFloat32(offset + 8);
+            let binCount = data.getUint32(offset + 12);
+            offset += 16;
+            for (let i = 0; i < binCount; i++, offset += 4) {
+                let binValue = data.getUint32(offset);
+                values.push({
+                    bin_start: i * binSize,
+                    bin_end: (i + 1) * binSize,
+                    count: binValue,
+                });
+            }
+
+            return {binSize, mean, median, values};
+        })
+        .then((data) => {
+            vl.data(data)
+                .title(`Number of gRNA per cell`)
+                .width(450)
+                .height(300)
+                .layer(
+                    vl
+                        .markBar({tooltip: true, color: "dimgray"})
+                        .encode(
+                            vl
+                                .x()
+                                .fieldQ("bin_start")
+                                .bin({step: data.binSize, binned: true})
+                                .title(`Number of gRNA per cell`),
+                            vl.x2().fieldQ("bin_end"),
+                            vl.y().fieldQ("count").title("Counts")
+                        ),
+                    vl
+                        .markRule({size: 3, color: "red", tooltip: {content: "encoding"}})
+                        .encode(vl.x().datum(data.median), vl.tooltip().value(`Median: ${data.median}`)),
+                    vl
+                        .markRule({size: 3, color: "yellow", tooltip: {content: "encoding"}})
+                        .encode(vl.x().datum(data.mean), vl.tooltip().value(`Mean: ${data.mean}`)),
+
+                    vl
+                        .markText({align: "center", baseline: "bottom", dx: 0, dy: -5})
+                        .encode(vl.x().datum(data.median), vl.y().value(0), vl.text().value("Median")),
+
+                    vl
+                        .markText({align: "center", baseline: "bottom", dx: 20, dy: 15})
+                        .encode(vl.x().datum(data.mean), vl.y().value(0), vl.text().value("Mean"))
+                )
+                .render()
+                .then((viewElement) => {
+                    // render returns a promise to a DOM element containing the chart
+                    // viewElement.value contains the Vega View object instance
+                    document.getElementById("ngrna_histogram").appendChild(viewElement);
+                });
```

Comment:
Since this code is mostly shared across histogram plots I think you should factor it out into a function, instead of repeating the same code in two places.